include/dump_offload.hpp

```diff
@@ -0,0 +1,445 @@
+                                 const std::string& dumpEntryType);
+inline void resetHandler();
+
+static constexpr size_t socketBufferSize = static_cast<size_t>(64 * 1024);
+static constexpr uint8_t maxConnectRetryCount = 3;
+
+/** class Handler
+ *  Handles data transfer between unix domain socket and http connection socket.
+ *  This handler invokes dump offload reads data from unix domain socket
+ *  and writes on to http stream connection socket.
+ */
+class Handler : public std::enable_shared_from_this<Handler>
+{
+  public:
+    Handler(boost::asio::io_context& ios, const std::string& entryIDIn,
+            const std::string& dumpTypeIn,
+            const std::string& unixSocketPathIn) :
+        entryID(entryIDIn),
+        dumpType(dumpTypeIn), unixSocketPath(unixSocketPathIn), unixSocket(ios),
+        waitTimer(ios)
+    {}
+
+    /**
+     * @brief Connects to unix socket to read dump data
+     *
+     * @return void
+     */
+    void doConnect()
+    {
+        this->unixSocket.async_connect(
+            unixSocketPath.c_str(), [this, self(shared_from_this())](
+                                        const boost::system::error_code& ec) {
+            if (ec)
+            {
+                // TODO:
+                // right now we don't have dbus method which can make sure
+                // unix socket is ready to accept connection so its possible
+                // that bmcweb can try to connect to socket before even
+                // socket setup, so using retry mechanism with timeout.
+                if (ec == boost::system::errc::no_such_file_or_directory ||
+                    ec == boost::system::errc::connection_refused)
+                {
+                    BMCWEB_LOG_DEBUG("UNIX Socket: async_connect {}", ec);
+                    retrySocketConnect();
+                    return;
+                }
+                BMCWEB_LOG_ERROR("UNIX Socket: async_connect error {}", ec);
+                waitTimer.cancel();
+                this->connection->sendStreamErrorStatus(
+                    boost::beast::http::status::internal_server_error);
+                this->connection->close();
+                this->cleanupSocketFiles();
+                return;
+            }
+            waitTimer.cancel();
+            this->connection->sendStreamHeaders(std::to_string(this->dumpSize),
+                                                "application/octet-stream");
+            this->doReadStream();
+        });
+    }
+
+    /**
+     * @brief  Invokes InitiateOffload method of dump manager which
+     *         directs dump manager to start writing on unix domain socket.
+     *
+     * @return void
+     */
+    void initiateOffload()
+    {
+        crow::connections::systemBus->async_method_call(
+            [this,
+             self(shared_from_this())](const boost::system::error_code& ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("DBUS response error: {}", ec);
```

Comment:
ideally this only logged on the internal error 


                if (ec.value() == EBADR)
                {
                    this->connection->sendStreamErrorStatus(
                        boost::beast::http::status::not_found);
                }
                else
                {
                BMCWEB_LOG_ERROR("DBUS response error: {}", ec);
                    this->connection->sendStreamErrorStatus(
                        boost::beast::http::status::internal_server_error);
                }

Reply:
done

Reply:
I think internal would be enough here

Reply:
but they are different, to the client  not_found means I am wrong. 
internal error is the server is wrong

Reply:
I think 

            if (ec.value() == EBADR)
            {
                this->connection->sendStreamErrorStatus(
                    boost::beast::http::status::not_found);
            }
            else
            {
            BMCWEB_LOG_ERROR("DBUS response error: {}", ec);
                this->connection->sendStreamErrorStatus(
                    boost::beast::http::status::internal_server_error);
            }
            
 makes the most sense

Reply:
done